### PR TITLE
#170868448 Add Book sections

### PR DESCRIPTION
**Pull request by hakoemmy**
#### What does this PR do?
 - Allows admin to add book sections
#### Description of Task to be completed?
- Have ``` /api/v1/sections ``` endpoint working
#### How should this be manually tested?
just clone the [repo](https://github.com/blaise82/l.m.s.git
) and run ``` /api/v1/sections ``` endpoint in postman
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170868448 ](https://www.pivotaltracker.com/story/show/170868448)
#### Screenshots (if appropriate)
Screen Shot 2020-01-27 at 22 12 26

#### Questions:
N/A

### CODE REVIEW

**Review by blaise82 — CHANGES_REQUESTED** [review]
you shouldn't be returning a **message** on fail instaid return an **error**
API Response Specification example
On Success
```
{
"status" : Integer: 201,
"message": String: "User created successfully"
"data" : Object: {...} or [{...}, {...}]
}
```
On Error
```
{
"status" : Integer: 401,
"error" : String: "Email already exist"
} 
```

otherwise, everything else looks great @hakoemmy 🔥

File: server/middleware/isUserAdmin.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    const  user = await User.findAll({
+      where: {
+        email: email.trim(),
+      },
+    });
+    if (user.length === 0) {
+      return res.status(401).json({
+        status: 401,
+        message: 'Awww, Snap!..Such kind of access token does not match any user!'
+      });
```
Comment (blaise82) [inline server/middleware/isUserAdmin.js]: as I mentioned before please try and return errors, not messages on failing
if possible change the error/message strings you are using to be more understandable and general for this middleware might be used again 
for example

_message: 'Awww, Snap!..Such kind of access token does not match any user!_
most of the users don't even know what a token is.
from my point of view even _**error**: "access denied"_ would be better 

may the same apply to all errors/messages to be returned

**Review by blaise82 — CHANGES_REQUESTED** [review]
messages/erros

File: server/middleware/isUserAdmin.js (hunk 1 above)
Comment (hakoemmy) [reply server/middleware/isUserAdmin.js]: @blaise82 thank you for such correction. I am on it

**Comment by AbdoulNuru** [comment]
@hakoemmy For me, everything looks fine but why did the coverage slightly drop?

**Comment by AbdoulNuru** [comment]
great🔥The coverage is back on track

**Comment by izzett222** [comment]
it looks good to me

**Review by blaise82 — APPROVED** [review]
good

**Review by izzett222 — APPROVED** [review]
it looks great to me

**Review by RusimbiPatrick — COMMENTED** [review]
You don't have to query the DB twice. There is a method you can use [findOrcreate](https://sequelize.readthedocs.io/en/2.0/docs/models-usage/#findorcreate-search-for-a-specific-element-or-create-it-if-not-available)

**Comment by hakoemmy** [comment]
@RusimbiP Could you please be specific? I mean cite what you mean in codebase

**Comment by RusimbiPatrick** [comment]
> @RusimbiP Could you please be specific? I mean cite what you mean in codebase
Merge it

**Review by RusimbiPatrick — APPROVED** [review]

**Pull request merged**